src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java

```diff
@@ -14,11 +14,13 @@
 public class StashBuildEnvironmentContributor extends EnvironmentContributor {
   @Override
   public void buildEnvironmentFor(
-      @Nonnull Run r, @Nonnull EnvVars envs, @Nonnull TaskListener listener)
+      @SuppressWarnings("rawtypes") @Nonnull Run r,
```

Comment:
Does this come from the super definition and cannot be changed? I'm surprised we have to suppress it in that case

Reply:
Yes, it comes from the superclass:
https://javadoc.jenkins.io/hudson/model/EnvironmentContributor.html#buildEnvironmentFor-hudson.model.Run-hudson.EnvVars-hudson.model.TaskListener-

It causes a warning with Java 8. it's not from FindBugs. Eclipse marks it as "java warning".

Reply:
I'm pretty sure that Eclipse has an option to ignore warnings caused by super-type signatures.
Does javac print out a warning?

Reply:
You are right, there is an option in Eclipse, it can be found from the tooltip on the warning by selecting "Configure problem severity". It's called "Ignore unavoidable generic type problems due to raw APIs".

The compiler will issue warnings if #50 is applied:

```
[WARNING] /home/roskinp/src/jenkins/plugins/stash-pullrequest-builder-plugin/src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java:[73,49] found raw type: hudson.model.AbstractProject
  missing type arguments for generic class hudson.model.AbstractProject<P,R>
[WARNING] /home/roskinp/src/jenkins/plugins/stash-pullrequest-builder-plugin/src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildEnvironmentContributor.java:[17,16] found raw type: hudson.model.Run
  missing type arguments for generic class hudson.model.Run<JobT,RunT>
```

But since we are not doing it, I think those suppressions can be removed. I think the Eclipse default settings are not good in that regard.

I'm glad I checked the warnings, it turns out the cast to `StashCause` in `StashBuildEnvironmentContributor.java` is redundant now.